File: fib_files/fib.php
<?php
function fib($n) {
    if ($n <= 2) {
        return 1;
    } else {
        return fib($n - 1) + fib($n - 2);
    }
}

$n_env = getenv("FIB_N");
$n = $n_env ? intval($n_env) : 40;

$sequence = [];
$start = microtime(true);

for ($i = 1; $i <= $n; $i++) {
    $val = fib($i);
    $sequence[] = $val;
    echo $i . ". fibonacci: " . $val . "\n";
}

$elapsed = microtime(true) - $start;

$result = [
    "language" => "PHP",
    "n" => $n,
    "sequence" => $sequence,
    "seconds" => $elapsed
];

file_put_contents("result_php.json", json_encode($result, JSON_PRETTY_PRINT));
?>

File: fib_files/fib.py
import os, json, time

def fib(n):
    if n <= 2:
        return 1
    return fib(n - 1) + fib(n - 2)

if __name__ == "__main__":
    n = int(os.getenv("FIB_N", "40"))

    start = time.time()
    sequence = []
    for i in range(1, n + 1):
        val = fib(i)
        sequence.append(val)
        print(f"{i}. fibonacci: {val}")
    elapsed = time.time() - start

    result = {
        "language": "Python",
        "n": n,
        "sequence": sequence,
        "seconds": elapsed
    }

    with open("result_python.json", "w", encoding="utf-8") as f:
        json.dump(result, f, indent=2)

